src/main/java/nu/studer/teamcity/buildscan/internal/ArtifactBuildScanDataStore.java

```diff
@@ -0,0 +1,87 @@
+package nu.studer.teamcity.buildscan.internal;
+
+import com.google.common.base.Charsets;
+import jetbrains.buildServer.ArtifactsConstants;
+import jetbrains.buildServer.serverSide.SBuild;
+import nu.studer.teamcity.buildscan.BuildScanReference;
+import org.apache.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public final class ArtifactBuildScanDataStore implements BuildScanDataStore {
+
+    private static final Logger LOGGER = Logger.getLogger("jetbrains.buildServer.BUILDSCAN");
+    public static final String BUILDSCANS_DIR = "buildscans";
```

Comment:
Is is worth to put the urls file into a folder with the same name or should we just add just under .teamcity?

Reply:
directly under .teamcity is probably fine as well. I just followed [the convention, that plugins should have their own directory]( 
https://plugins.jetbrains.com/docs/teamcity/storing-plugin-state-and-settings.html#StoringPluginStateandSettings-Build-levelState).

Reply:
Then let's keep as it currently is.